### PR TITLE
Automatically fills import statement and display autocomplete popup

### PR DESCRIPTION
**Pull request by aliev**
there is a short screencast example: https://dl.dropboxusercontent.com/u/5837324/autoimport.mov

### CODE REVIEW

**Comment by davidhalter** [comment]
I'm still not 100% sure if it makes sense to include this. What do you think @blueyed?

File: autoload/jedi.vim [hunk 1]
```diff
@@ -222,6 +222,16 @@ function! jedi#force_py_version_switch()
 endfunction
 
 
+function! jedi#auto_put_import()
+    " Auto put import statement after from module.name<space> and complete
+    if search('\<from\s\+[A-Za-z0-9._]\+\s*\%#\s*$', 'bcn', line('.'))
```
Comment (davidhalter) [inline autoload/jedi.vim]: Shouldn't we check for whitespace before `from`? wouldn't that make sense?

**Comment by blueyed** [comment]
I like the general idea!

But it will map "<Space>" and therefore cause some overhead for _every_ space you type.
The pattern could be improved, e.g. by anchoring it with `'^\s\+`, and there should be probably a fast-path handling, which is cheaper.

And then it's just some kind of basic snippet.
E.g. with UltiSnips you have a "from" snippet already, which expands to `from ${1:module} import ${2:Stuff}`.
When using this snippet, it would be good if there was a `smap` like this:

    smap <c-@> <del><C-x><C-o><C-r>=jedi#complete_opened()<CR>

Then you can do the following (<C-j> being my UltiSnips trigger):

    from<C-j>os<C-j><C-space>

I think this `:smap` is useful to have anyway - I'll add a PR for this.

File: autoload/jedi.vim (hunk 1 above)
Comment (blueyed) [inline autoload/jedi.vim]: The `\<` is a word boundary, but not necessarily whitespace, yes.

I would also anchor it at the beginning: `^\s\+

**Comment by davidhalter** [comment]
Shouldn't we create an option that's more generic? Instead of `let g:jedi#auto_put_import = 1` we could use `jedi#auto_smart_completion` (which might be confusing, but something like it). Because basically you could add such checks for other things as well. (I don't have any good examples right now). Ideas?

**Comment by blueyed** [comment]
:+1: for another name.
I suggest `jedi#smart_auto_mappings`.

**Comment by aliev** [comment]
@davidhalter good idea!

**Comment by davidhalter** [comment]
:+1:

**Comment by blueyed** [comment]
@aliev 
Will you update the PR?
What do you think about having a fast-as-possible code path - at least by anchoring it to the beginning of the line?
I think the default for this should be 0 for now, while it's experimental.

**Comment by davidhalter** [comment]
I actually think the default should be 1. It's not a very complicated thing and should just work. If it doesn't we shouldn't even merge.

**Comment by aliev** [comment]
@blueyed @davidhalter oh sorry for big delay with my answer. I'm beginner in writing of vim of scripts. whether it is necessary to write some tests?

**Comment by davidhalter** [comment]
So what I would like is:

1. Squash the commits (there are 3 `added doc`).
2. Use `jedi#smart_auto_mappings` as a name instead of `jedi#auto_put_import`.
3. Set the default to 1
4. A test would be nice. But you don't have to.

**Pull request closed without merging**